api/base/views.py

```diff
@@ -129,7 +129,7 @@ def get_serializer_context(self):
         context.update({
             'enable_esi': (
                 is_truthy(self.request.query_params.get('esi', django_settings.ENABLE_ESI)) and
-                self.request.accepted_media_type in django_settings.ESI_MEDIA_TYPES
+                self.request.accepted_renderer.media_type in django_settings.ESI_MEDIA_TYPES
```

Comment:
This, BTW, is why the varnish part of the travis tests were passing. Since self.request.accepted_media_type decided to stop existing sometime between 3.1.1 and 3.3.3 ESI was being disabled by the varnish tests even though it was being enabled through the setting and the correct content type. So the varnish tests were comparing data where ESI was false against data where ESI was false and passing. @brianjgeiger @icereval @Ghalko